File: e2e/test-resources/android-project/src/test/java/com/example/myapplication/MainActivityTest.java
package com.example.myapplication;

import android.widget.TextView;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.android.controller.ActivityController;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class MainActivityTest {
    @Test
    public void testTextView() {
        try (ActivityController<MainActivity> controller = Robolectric.buildActivity(MainActivity.class)) {
            controller.setup();
            MainActivity activity = controller.get();
            TextView textView = activity.findViewById(R.id.text_view);
            assertEquals("Search", textView.getText());
        }
    }
}

File: e2e/test-resources/android-kotlin-project/src/test/java/com/example/myapplication/MainActivityTest.kt
package com.example.myapplication

import android.widget.TextView
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class MainActivityTest {
    @Test
    fun testTextView() {
        Robolectric.buildActivity(MainActivity::class.java).use { controller ->
            controller.setup()
            val activity = controller.get()
            val textView = activity.findViewById<TextView>(R.id.text_view)
            assertEquals("Search", textView.text)
        }
    }
}
